### PR TITLE
Fix generating duplicates and omitting heading

### LINKED ISSUE #2012: Add missing 'Source Code' heading in `krel` output
#### What happened:
'Source Code' heading is missing in `krel`-generated files.
Please refer to https://github.com/kubernetes/kubernetes/pull/101299/files .

[As-is]
```Markdown
## Downloads for v1.21.0

filename | sha512 hash
-------- | -----------
```

#### What you expected to happen:

[To-be]

```Markdown
## Downloads for v1.21.0

### Source Code

filename | sha512 hash
-------- | -----------
```

#### How to reproduce it (as minimally and precisely as possible):
Run `krel changelog` and/or `krel release-notes` commands, maybe..?

#### Anything else we need to know?:
- Related issue: #1993
- Related issue: https://github.com/kubernetes/sig-release/issues/1528
- CC: @wilsonehusin @pmmalinov01 

#### Environment:

- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Others:

### PR DESCRIPTION
**Pull request by jihoon-seo**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

- (#1993: "Remove duplicates in `krel release-notes` output")
> Having headers in `krel release-notes` output caused the end result of k/k changelog to have duplicated headers. We had to fix this in 1.20 and again in 1.21 after the release.
- (#2012: "Add missing 'Source Code' heading in `krel` output")
> 'Source Code' heading is missing in krel-generated files.
Please refer to https://github.com/kubernetes/kubernetes/pull/101299/files .

#### Which issue(s) this PR fixes:

- Fixes a part of #1993 ("Remove duplicates in `krel release-notes` output")
- Fixes #2012 ("Add missing 'Source Code' heading in `krel` output")

#### Does this PR introduce a user-facing change?

```release-note
pkg/changelog: Add missing "Source Code" heading, fix upper-case consistency in headings, and remove duplicates
```

### CODE REVIEW

**Comment by jihoon-seo** [comment]
CC: @wilsonehusin , PTAL!

**Comment by wilsonehusin** [comment]
this is awesome, thanks for contributing!

I think this warrants a release note? something like:

```
pkg/changelog: Add missing "downloads" heading and fix upper-case consistency in headings
```

otherwise, code is 👍🏼 

/lgtm

**Comment by jihoon-seo** [comment]
@wilsonehusin Thank you for your suggestion.

I propose another suggestion which is almost the same:
```
pkg/changelog: Add missing "Source Code" heading, fix upper-case consistency in headings, and remove duplicates
```

Would it be okay to put this release note into the "Does this PR introduce a user-facing change?" field of this PR? 😊

**Comment by wilsonehusin** [comment]
ah good catch -- yes that looks better!

> Would it be okay to put this release note into the "Does this PR introduce a user-facing change?" field of this PR? 😊

yep that's the right place to do it

**Comment by jihoon-seo** [comment]
@wilsonehusin Done putting. PTAL. Thank you! 😊

**Comment by wilsonehusin** [comment]
let's wait for people with approval powers, i don't think i have any here 😆 

/priority backlog
/milestone v1.22

File: pkg/notes/document/testdata/document.md.golden
Comment (saschagrunert) [inline pkg/notes/document/testdata/document.md.golden:3]: I would fix this link to https://docs.k8s.io and maybe add a better explanation like:

> The latest documentation for Kubernetes can be found on [docs.k8s.io](https://docs.k8s.io)

Comment (wilsonehusin) [reply pkg/notes/document/testdata/document.md.golden:3]: I think this is meant to be removed because the original release notes generated by krel for patch releases doesn't have this section.

Or is it meant to be different?

Comment (saschagrunert) [reply pkg/notes/document/testdata/document.md.golden:3]: Ah, got it. :+1:

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**
